### ISSUE TITLE
Multiple "resource does not have attribute" errors on destroy run

### ISSUE THREAD

**Issue by rcgenova** [hashicorp/terraform#8869]
### Terraform Version

0.7.2
### Affected Resource(s)

atlas_artifact
google_compute_network
### Debug Output

Atlas run links:
https://atlas.hashicorp.com/terraform/rcgenova/environments/google-us-central1-prod/changes/runs/66
https://atlas.hashicorp.com/terraform/rcgenova/environments/google-us-central1-prod/changes/runs/68 (TF_LOG=DEBUG)
### Actual Behavior

Received the following errors at the end of the apply/destroy:

```
Error applying plan:

6 error(s) occurred:

* Resource 'data.atlas_artifact.google-ubuntu-haproxy' does not have attribute 'id' for variable 'data.atlas_artifact.google-ubuntu-haproxy.id'
* Resource 'data.atlas_artifact.google-ubuntu-nodejs' does not have attribute 'id' for variable 'data.atlas_artifact.google-ubuntu-nodejs.id'
* Resource 'data.atlas_artifact.google-ubuntu-consul' does not have attribute 'id' for variable 'data.atlas_artifact.google-ubuntu-consul.id'
* Resource 'data.atlas_artifact.google-ubuntu-vault' does not have attribute 'id' for variable 'data.atlas_artifact.google-ubuntu-vault.id'
* Resource 'google_compute_network.network' does not have attribute 'self_link' for variable 'google_compute_network.network.self_link'
* Resource 'google_compute_network.network' does not have attribute 'self_link' for variable 'google_compute_network.network.self_link'
```
### References

Possibly related to: https://github.com/hashicorp/terraform/issues/6991

**Comment by jen20** [comment]
@jbardin I wonder if this is related to one of the races you have found recently?

**Comment by rwcee** [comment]
Confirming this is an issue in 0.7.4.

**Comment by mitchellh** [comment]
This should be fixed in 0.8.0 beta.
